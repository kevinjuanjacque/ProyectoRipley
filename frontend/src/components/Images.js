import { useContext, useState } from 'react';
import fetchProduct from '../helpers/FetchProduct';
import { UseContext } from '../hooks/useContext';

export const Images = () => {
    
    const resultado=fetchProduct.FetchImagenesProduct();
    const img = resultado.resp;

    const {productos,setproductos} = useContext(UseContext)

    const [Photo, setPhoto] = useState(0)

    const NextPhoto = ()=>{
        if(Photo === img.length-1){
            setPhoto(0);
            return ;
        }
        setPhoto(Photo+1);
    }

    const addProduct=()=>{
        
        setproductos([...productos,'191919']);

    }

    return (
        <div>
            {( resultado.loading ) ? <div className="spinner-border" role="status">
                                        <span className="sr-only">Loading...</span>
                                    </div> :
                                    (<div height="500"><img className="img" src={`https:${img[Photo]}`} height="480" onClick={NextPhoto} alt="imagenesProduct" />
                                    <button className="btn btn-primary btn-block mt-3 ml-2" onClick={addProduct} > Agregar a la bolsa </button></div>)
                                    }
        </div>
    )
}
