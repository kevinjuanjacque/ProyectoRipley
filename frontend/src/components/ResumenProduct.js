
import fetchProduct from '../helpers/FetchProduct';
import { Envio } from './resumen/Envio';
import { Garantia } from './resumen/Garantia';
import { Headers } from './resumen/Headers';
import { Prices } from './resumen/Prices';
import { PuntosRipley } from './resumen/PuntosRipley';

export const ResumenProduct = () => {
    const resultado=fetchProduct.FetchResumenProduct();
    const {name,sku,shortDescription,prices,ripleyPuntos,shipping,garantia}=resultado.resp;
   
    return (
        <div>
            {(resultado.loading) ? (<div className="spinner-border" role="status">
                                        <span className="sr-only">Loading...</span>
                                    </div>) :
                                    (
                                        <>
                                            <Headers name={name} sku={sku} shortDescription={shortDescription} />

                                            <Prices prices={prices}/>

                                            <PuntosRipley ripleyPuntos={ripleyPuntos} />

                                            
                                            <Envio shipping={shipping } />

                                            <Garantia garantia={garantia} />
                                        </>
                                    )
            }
        </div>
    )
}
