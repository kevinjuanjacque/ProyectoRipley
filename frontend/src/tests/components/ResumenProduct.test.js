import {shallow} from 'enzyme'
import { ResumenProduct } from '../../components/resumenProduct'
describe('Prueba en componente <ResumenProduct />', () => {
    test('prueba renderizado correcto', () => {

        const wrapper=shallow(<ResumenProduct />);

        expect(wrapper).toMatchSnapshot();
        
    });
    
    
})
