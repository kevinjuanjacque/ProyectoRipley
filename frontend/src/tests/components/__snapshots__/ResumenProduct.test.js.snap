// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en componente <ResumenProduct /> prueba renderizado correcto 1`] = `
<div>
  <div
    className="spinner-border"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
